pubsub/pubsub.go

```diff
@@ -361,9 +361,10 @@ const (
 	// by more than that.
 	maxShrinkFactor = 0.75
 
-	// The maximum batch size to request.
-	// TODO(rvangent): Increase this.
-	maxBatchSize = 1000
+	// The maximum batch size to request. Setting this too low doesn't allow
+	// drivers to get lots of messages at once; setting it too small risks having
+	// drivers spend a long time in ReceiveBatch trying to achieve it.
+	maxBatchSize = 3000
```

Comment:
Where is this used?

Reply:
Line 407:

```
s.runningBatchSize = math.Min(s.runningBatchSize, maxBatchSize)
```

It used to be hard-coded to 1000 with a long comment about how GCP Pub/Sub returns a max of 1000.